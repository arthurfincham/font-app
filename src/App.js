import { useState, useRef, useEffect } from 'react';
import PreviewTable from './components/PreviewTable';
import Navbar from './components/Navbar';
import { ThemeProvider } from 'styled-components';
import { createTheme } from '@mui/material/styles';
import { GlobalStyles } from './global';
import { SizeProvider } from './context/SizeContext';
import { WeightProvider } from './context/WeightContext';
import { ItalicProvider } from './context/ItalicContext';
import { CodeSnippetProvider } from './context/CodeSnippetContext';

function App() {
  const [previewText, setPreviewText] = useState([null]);

  const previewInput = useRef(null);

  const updatePreviewText = (e) => {
    setPreviewText(e.target.value);
  };

  const [bodyColor, setBodyColor] = useState('#FFF');
  const [textColor, setTextColor] = useState('#000');

  const colorTheme = createTheme({
    body: bodyColor,
    text: textColor,
    breakpoints: {
      values: {
        xxs: 0, // small phone
        xs: 300, // phone
        sm: 600, // tablets
        md: 900, // small laptop
        lg: 1200, // desktop
        xl: 1536, // large screens
      },
    },
  });

  const navProps = {
    setBodyColor,
    bodyColor,
    setTextColor,
    textColor,
    updatePreviewText,
    previewText,
    previewInput,
  };

  useEffect(() => {
    previewInput.current.dispatchEvent(
      new Event('change', {
        detail: {
          newValue: previewText,
        },
        bubbles: true,
        cancelable: true,
      })
    );
  }, [previewText]);

  return (
    <>
      <CodeSnippetProvider>
        <ItalicProvider>
          <WeightProvider>
            <SizeProvider>
              <Navbar {...navProps} />
              <main>
                <ThemeProvider theme={colorTheme}>
                  <GlobalStyles />
                  <PreviewTable previewText={previewText} sx={{ height: '100%' }} />
                </ThemeProvider>
              </main>
            </SizeProvider>
          </WeightProvider>
        </ItalicProvider>
      </CodeSnippetProvider>
    </>
  );
}

export default App;
