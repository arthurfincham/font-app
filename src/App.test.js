import App from './App';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({ adapter: new Adapter() });

it('renders App correctly', () => {
  const wrapper = shallow(<App />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
