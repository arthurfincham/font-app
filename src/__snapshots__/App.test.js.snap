// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<AppContextProvider>
  <Navbar
    previewInput={
      Object {
        "current": null,
      }
    }
    previewText={
      Array [
        null,
      ]
    }
    updatePreviewText={[Function]}
  />
  <main>
    <PreviewTable
      previewText={
        Array [
          null,
        ]
      }
      sx={
        Object {
          "height": "100%",
        }
      }
    />
  </main>
</AppContextProvider>
`;
